src/utilities/git.jl

```diff
@@ -1,5 +1,78 @@
+const COMPATHELPER_GIT_COMMITTER_NAME = "CompatHelper Julia"
+const COMPATHELPER_GIT_COMMITTER_EMAIL = "compathelper_noreply@julialang.org"
+
+function get_git_name_and_email(; env=ENV)
+    name = if "GIT_COMMITTER_NAME" in env
+        env["GIT_COMMITTER_NAME"]
+    else
+        COMPATHELPER_GIT_COMMITTER_NAME
+    end
+
+    email = if "GIT_COMMITTER_EMAIL" in env
+        env["GIT_COMMITTER_EMAIL"]
+    else
+        COMPATHELPER_GIT_COMMITTER_EMAIL
+    end
+
+    return name, email
+end
+
 git_checkout(branch::AbstractString) = run(`git checkout $(branch)`)
 
-function git_clone(url::AbstractString, local_path::AbstractString)
-    return run(`git clone $(url) $(local_path)`)
+git_add(; items="", flags="") = run(`git add $flags $items`)
+
+git_remote_remove(remote::AbstractString) = run(`git remote remove $remote`)
+function git_remote_add(remote::AbstractString, url::AbstractString)
+    return run(`git remote add $remote $url`)
+end
+
+git_reset(pathspec::AbstractString; flags="") = run(`git reset $flags "$pathspec"`)
+
+function git_push(remote::AbstractString, branch::AbstractString; force=false, env=ENV)
+    force_flag = force ? "-f" : ""
+    name, email = get_git_name_and_email(; env=env)
+
+    return run(`git -c "user.name=$name" -c "user.email=$email" push $force_flag $remote $branch`)
+end
+
+function git_commit(
+    message::AbstractString="",
+    pkey_filename::Union{AbstractString,Nothing}=nothing;
+    env=ENV,
+)
+    name, email = get_git_name_and_email(; env=env)
+    cmd = `git -c "user.name=$name" -c "user.email=$email" commit -m $message`
+
+    result = try
+        withenv("GIT_SSH_COMMAND" => isnothing(pkey) ? "ssh" : "ssh -i $pkey_filename")do
+            p = pipeline(cmd; stdout=stdout, stderr=stderr)
+            success(p)
+        end
+    catch
+        false
+    end
+    return result
+end
+
+function git_branch(branch::AbstractString; checkout=false)
+    run(`git branch $(branch)`)
+    checkout && git_checkout(branch)
```

Comment:
```suggestion
    return checkout && git_checkout(branch)
```

Reply:
Do we need this to return? I don't think it returns anything

Reply:
It'll return regardless I believe? Also conforms to BlueStyle to have a return in non-inline functions.

Reply:
We can just `return nothing ` at the end.

Reply:
yeah I mean it will return either way, but there's nothing that we necessarily care about it returning? We won't use the result of that call. I think Bluestyle always wants an explicit return if you are actually returning something you will be using, rather than a function that just does a thing without actually returning anything we want

Reply:
Tbh I can go either way on it, doesn't really matter to me. Even if it returns nothing I personally like it being explicit, but it's no worries either way.

Reply:
yeah thats fine. Bluestyle wants you to return `something`. I disagree, but I'll conform to it